r/R/RecordBatch.R

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#' @include R6.R
+
+`arrow::RecordBatch` <- R6Class("arrow::RecordBatch", inherit = `arrow::Object`,
+  public = list(
+    num_columns = function() RecordBatch__num_columns(self),
+    num_rows = function() RecordBatch__num_rows(self),
+    schema = function() `arrow::Schema`$new(RecordBatch__schema(self)),
+    to_file = function(path) invisible(RecordBatch__to_file(self, fs::path_abs(path))),
```

Comment:
Maybe `to_ipc_file` to be more explicit. We'll want to implement methods like `pyarrow.read_record_batch` that deals with more granular reads from IO sources in the future